gstreamer-source.c

```diff
@@ -316,6 +367,86 @@ const char *gstreamer_source_get_name(void *type_data)
 	return "GStreamer Source";
 }
 
+enum obs_media_state gstreamer_source_get_state(void *user_data)
+{
+	data_t *data = user_data;
+
+	return data->obs_media_state;
+}
+
+int64_t gstreamer_source_get_time(void *user_data)
+{
+	data_t *data = user_data;
```

Comment:
I wonder if all of these should check again data->pipe not being NULL.

Reply:
Yup.  [v4] adds the missing `if (data->pipe)` guards to the media callbacks that should have them.